plugins/integration/providers/voicebase/lib/kVoicebaseFlowManager.php

```diff
@@ -0,0 +1,165 @@
+			}
+			$transcript->setStatus(AttachmentAsset::ASSET_STATUS_QUEUED);
+			$transcript->save();
+	
+			return true;
+		}
+	
+		$formatsString = $providerData->getCaptionAssetFormats();		
+	
+		if($dbBatchJob->getStatus() == BatchJob::BATCHJOB_STATUS_FINISHED)
+		{
+			$key = $providerData->getApiKey();
+			$password = $providerData->getApiPassword();
+			$serviceProviderParams = array('apiKey' => $key , 'apiPassword' => $password);
+			$clientHelper = VoicebasePlugin::getClientHelper($serviceProviderParams);
+		
+			$externalEntryExists = $clientHelper->checkExitingExternalContent($entryId);
+			if (!$externalEntryExists)
+			{
+				KalturaLog::err('remote content does not exist');
+				return true;     	
+			}
+			$formatsArray = explode(',',$formatsString);
+			$formatsArray[] = "TXT";
+			$contentsArray = $clientHelper->getRemoteTranscripts($entryId, $formatsArray);
+			KalturaLog::debug('contents are - ' . print_r($contentsArray, true));
+			$transcript = $this->getObjects($entryId, array(TranscriptPlugin::getAssetTypeCoreValue(TranscriptAssetType::TRANSCRIPT)), $spokenLanguage, true);
+			$captions = $this->getObjects($entryId, array(CaptionPlugin::getAssetTypeCoreValue(CaptionAssetType::CAPTION)), $spokenLanguage);
+		
+			$this->setObjectContent($transcript, $contentsArray["TXT"]);
+			unset($contentsArray["TXT"]);
+	
+			foreach ($contentsArray as $format => $content)
+			{        
+				$captionFormatConst = constant("KalturaCaptionType::" . $format);
+				if(isset($captions[$captionFormatConst]))
+					$caption = $captions[$captionFormatConst];
+				else
+				{
+					$caption = new CaptionAsset();
+					$caption->setEntryId($entryId);
+					$caption->setPartnerId($partnerId);
+					$caption->setLanguage($spokenLanguage);
+					$caption->setContainerFormat($captionFormatConst);
+					$caption->setAccuracy(self::DEFAULT_ACCURACY);
+					$caption->setStatus(CaptionAsset::ASSET_STATUS_QUEUED);
+					$caption->save();
+				}
+				$this->setObjectContent($caption, $content, $format);
+			}
+		}
+		return true;					    
+	}
+	
+	function getObjects($entryId, array $assetTypes, $spokenLanguage = null, $returnSingleObject = false)
+	{
+		$objects = $returnSingleObject ? null : array();
+		$statuses = array(asset::ASSET_STATUS_QUEUED, asset::ASSET_STATUS_READY);
+		$resultArray = assetPeer::retrieveByEntryId($entryId, $assetTypes, $statuses);
+	
+		foreach($resultArray as $object)
+		{
+			if($spokenLanguage)
+			{
+				if($object->getLanguage() == $spokenLanguage)
+				{
+					if ($returnSingleObject)
+						return $object;
+					$objects[$object->getContainerFormat()] = $object;
+				}
+			}
+			elseif($returnSingleObject)
+				return $object;
+			else
+				$objects[$object->getContainerFormat()] = $object;
+		}	
+		return $objects;
+	}
+	
+	private function setObjectContent($assetObject, $content, $format = null)
+	{
+		file_put_contents($fname = tempnam(myContentStorage::getFSUploadsPath(), "VBF"), $content); //prefix for - voicebase file
```

Comment:
Instead of calling file_put_contents and then kFileSyncUtils::moveFromFile, please call only kFileSyncUtils::file_put_contents.